src/components/uikit/BottomSheet.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useCallback, useRef, ReactNode } from 'react';
+import { Dimensions } from 'react-native';
+import RNBottomSheet, {
+  BottomSheetProps as RNBottomSheetProps,
+} from '@gorhom/bottom-sheet';
+
+import { styled, useTheme } from '~styles';
+
+interface BottomSheetProps {
+  initialIndex: number;
+  snapPoints?: string[];
+  children: ReactNode;
+  onSheetChanges?: (index: number) => void;
+  keyboardBehavior?: RNBottomSheetProps['keyboardBehavior'];
+}
+
+export function BottomSheet({
+  initialIndex,
+  snapPoints = ['25%', '50%'],
+  children,
+  onSheetChanges,
+  keyboardBehavior = 'interactive',
+}: BottomSheetProps) {
+  const theme = useTheme();
+  const bottomSheetRef = useRef<RNBottomSheet>(null);
+
+  const handleSheetChanges = useCallback(
```

Comment:
This memoization is kindah useless as the reference for `onSheetChanges` will most likely be different on each render unless the parent component memoizes it which it probably wont' do and shouldn't need to do. Do we even need this memoization with useCallback here?

Reply:
Good point. I took it from the docs: https://ui.gorhom.dev/components/bottom-sheet/modal/usage

Reply:
Yeah the docs for that lib are ridiculous tbh 😄 They overuse memoization like crazy.

Reply:
Let's remove this memoization.